sql/service/src/main/scala/org/apache/spark/sql/service/cli/session/SessionManager.scala

```diff
@@ -0,0 +1,346 @@
+    }
+    cleanupLoggingRootDir()
+  }
+
+  private def cleanupLoggingRootDir(): Unit = {
+    if (isOperationLogEnabled) {
+      try {
+        FileUtils.forceDelete(operationLogRootDir)
+      } catch {
+        case e: Exception =>
+          logWarning("Failed to cleanup root dir of HS2 logging: " +
+            operationLogRootDir.getAbsolutePath, e)
+      }
+    }
+  }
+
+  @throws[ServiceSQLException]
+  def openSession(protocol: TProtocolVersion, username: String,
+                  password: String, ipAddress: String,
+                  sessionConf: util.Map[String, String]): SessionHandle = {
+    openSession(protocol, username, password, ipAddress,
+      sessionConf, false, null)
+  }
+
+  /**
+   * Opens a new session and creates a session handle.
+   * The username passed to this method is the effective username.
+   * If withImpersonation is true (==doAs true) we wrap all the calls in ServiceSession
+   * within a UGI.doAs, where UGI corresponds to the effective user.
+   *
+   * Please see {@code ThriftCLIService.getUserName()} for
+   * more details.
+   *
+   * @param protocol
+   * @param username
+   * @param password
+   * @param ipAddress
+   * @param sessionConf
+   * @param withImpersonation
+   * @param delegationToken
+   * @return
+   * @throws ServiceSQLException
+   */
+  @throws[ServiceSQLException]
+  def openSession(protocol: TProtocolVersion, username: String, password: String,
+                  ipAddress: String, sessionConf: util.Map[String, String],
+                  withImpersonation: Boolean, delegationToken: String): SessionHandle = {
+    logWarning("create new session context")
+    logWarning(s"sqlContext = ${sqlContext}")
+    val ctx = if (sqlContext.conf.hiveThriftServerSingleSession) {
+      logWarning("single session")
+      sqlContext
+    } else {
+      logWarning("No single session")
+      sqlContext.newSession()
+    }
+    if (sessionConf != null && sessionConf.containsKey("use:database")) {
+      ctx.sql(s"use ${sessionConf.get("use:database")}")
+    }
+
+    var session: ServiceSession = null
+    // If doAs is set to true for SparkServer2, we will create a proxy object for the session impl.
+    // Within the proxy object, we wrap the method call in a UserGroupInformation#doAs
+    if (withImpersonation) {
+      val sessionWithUGI = new ServiceSessionImplwithUGI(protocol, username, password,
+        ctx, ipAddress, delegationToken)
+      session = ServiceSessionProxy.getProxy(sessionWithUGI, sessionWithUGI.getSessionUgi)
+      sessionWithUGI.setProxySession(session)
+    } else session = new ServiceSessionImpl(protocol, username, password, ctx, ipAddress)
+    session.setSessionManager(this)
+    session.setOperationManager(operationManager)
+    try {
+      session.open(sessionConf)
+    } catch {
+      case e: Exception =>
+        try {
+          session.close()
+        } catch {
+          case t: Throwable =>
+            logWarning("Error closing session", t)
```

Comment:
Add more info to the message - e.g. session id?

Reply:
> Add more info to the message - e.g. session id?

Since there won't return sessionHandle to client. Seems make no sense to print session id here